File: 序号/83.删除排序链表中的重复元素.cpp
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
      ListNode dummy(-101,head),*l=&dummy,*f=head;
      while(f){
        if(f->val==l->val){
          f=f->next;
        }else{
          l->next=f;
          l=f;f=f->next;
        }
      }
      l->next=nullptr;
      return dummy.next;
    }
};

File: 序号/83. 删除排序链表中的重复元素.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        p=fakeHead = ListNode(-101,head)
        while p.next:
            if p.val == p.next.val:
                p.next=p.next.next
            else:
                p=p.next
        return fakeHead.next